eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxo.hs

```diff
@@ -194,6 +202,24 @@ feesOK pp tx (UTxO utxo) =
             validateTotalCollateral pp txBody utxoCollateral
         ]
 
+-- | Test that inputs and refInpts are disjoint, in Conway and later Eras.
+disjointRefInputs ::
+  forall era.
+  EraPParams era =>
+  PParams era ->
+  Set (TxIn (EraCrypto era)) ->
+  Set (TxIn (EraCrypto era)) ->
+  Test (BabbageUtxoPredFailure era)
+disjointRefInputs pp inputs refInputs =
+  when
+    (pvMajor (pp ^. ppProtocolVersionL) > eraProtVerHigh @(BabbageEra (EraCrypto era)))
+    ( case nonEmpty (Set.toList common) of -- Nothing if common is empty
+        Just ne -> failureIf (not (Set.null common)) (BabbageNonDisjointRefInputs ne)
+        Nothing -> pure ()
+    )
```

Comment:
We now have `failureOnNonEmpty` that will be more efficient and will not force the computation when validation is turned off:


```suggestion
    (failureOnNonEmpty common BabbageNonDisjointRefInputs)
```

See the definition: https://github.com/IntersectMBO/cardano-ledger/blob/b05ba3609910c5e54ea0a27acf93dcd9a5dcd223/libs/small-steps/src/Control/State/Transition/Extended.hs#L413-L414